### PR TITLE
CFE-3047: Fix crash when calling string_split on something long

### PR DESCRIPTION
**Pull request by olehermanse**

### CODE REVIEW

**Comment by olehermanse** [comment]
@cf-bottom I want to see this fail in jenkins before I add the fix

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/2435/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/2435/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-2435/

**Comment by olehermanse** [comment]
@cf-bottom Alright, now let's test in jenkins that my fix makes it green!

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/2436/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/2436/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-2436/

File: tests/acceptance/01_vars/02_functions/string_split_2k_variable.cf [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+    "test_list" -> { "CFE-3047" } # Promise to ticket that we don't crash
+      slist => string_split("$(test_str)", "\n", "10000" );
+
+  reports:
+    any::
+      "$(this.promise_filename) Pass";
```
Comment (vpodzime) [inline tests/acceptance/01_vars/02_functions/string_split_2k_variable.cf]: I think it this should at least have a comment showing what the expected result is.

File: libpromises/rlist.c [hunk 2]
```diff
@@ -1195,9 +1195,9 @@ Rlist *RlistFromRegexSplitNoOverflow(const char *string, const char *regex, int
     while (count < max - 1 &&
            StringMatchWithPrecompiledRegex(pattern, sp, &start, &end))
     {
-        assert(start < CF_MAXVARSIZE);
-        memcpy(node, sp, start);
-        node[start] = '\0';
+        const size_t len = (start < CF_MAXVARSIZE) ? start : (CF_MAXVARSIZE - 1);
+        memcpy(node, sp, len);
+        node[len] = '\0';
```
Comment (vpodzime) [inline libpromises/rlist.c]: I think this should produce a warning.

**Comment by olehermanse** [comment]
Cherry-pick: https://github.com/cfengine/core/pull/3662

**Pull request merged**